File: src/main/java/com/example/provamiavuota/entities/Prodotto.java
package com.example.provamiavuota.entities;


import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import java.util.List;

@Getter
@Setter
@ToString
@EqualsAndHashCode
@Entity
@Table(name="prodotto",schema="e_commerce_milan",uniqueConstraints = {
        @UniqueConstraint(columnNames = {"nome", "categoria"})
})
public class Prodotto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id_prodotto",nullable = false)
    private int id;

    @Version
    @Column(name="version",nullable = false)
    @JsonIgnore
    private int version;

    @Basic
    @Column(name="nome",nullable = true,unique = true)
    private String nome;

    @Basic
    @Column(name="categoria",nullable = true)
    private String categoria;

    @Basic
    @Column(name="descrizione",nullable = true)
    private String descrizione;

    @Basic
    @Column(name="prezzo",nullable = true)
    private Double prezzo;

    @Basic
    @Column(name="quantita",nullable = true)
    private int quantita;

    @Basic
    @Column(name="hidden",length = 1)
    private int nascosto;


    @OneToMany(targetEntity = DettaglioOrdine.class, mappedBy="prodotto",cascade = CascadeType.MERGE)
    @JsonIgnore//cosi evito i cicli
    @ToString.Exclude
    private List<DettaglioOrdine> dettaglioOrdini;

    @OneToMany(targetEntity = ProdottiPromo.class, mappedBy="prodotto",cascade = CascadeType.MERGE)
    @JsonIgnore//cosi evito i cicli
    @ToString.Exclude
    private List<ProdottiPromo> dettaglioPromozioni;
}

File: frontend/lib/model/objects/Prodotto.dart
class Prodotto {
  int id;
  String nome;
  String categoria;
  String descrizione;
  double prezzo;
  int quantita;
  int nascosto;


  Prodotto({required this.id, required this.nome, required this.categoria, required this.descrizione, required this.prezzo,required this.quantita,required this.nascosto});

  factory Prodotto.fromJson(Map<String, dynamic> json) {
    return Prodotto(
      id: json['id'],
      nome: json['nome'],
      descrizione: json['descrizione'],
      categoria: json['categoria'],
      quantita: json['quantita'],
      prezzo: json['prezzo'],
      nascosto: json['nascosto']
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'nome': nome,
    'descrizione': descrizione,
    'categoria':categoria,
    'quantita': quantita,
    'prezzo': prezzo,
    'nascosto': nascosto
  };

  @override
  String toString() {
    return nome;
  }


}